cyclops-ctrl/internal/cluster/k8sclient/client.go

```diff
@@ -463,6 +465,23 @@ func (k *KubernetesClient) mapDeployment(group, version, kind, name, namespace s
 	}, nil
 }
 
+func (k *KubernetesClient) mapDaemonSet(group, version, kind, name, namespace string) (*dto.DaemonSet, error) {
+	daemonSet, err := k.clientset.AppsV1().DaemonSets(namespace).Get(context.Background(), name, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO pods handling
```

Comment:
You can do the same thing we did for Deployment pods [here](https://github.com/cyclops-ui/cyclops/blob/ba0a74443cfb82ffc05bf9ba58b22a4d27fbb1ee/cyclops-ctrl/internal/cluster/k8sclient/modules.go#L307)

This function accepts a Deployment and returns all pods belonging to that Deployment. You can create a new function similar to the one linked that will fetch pods for a DaemonSet. You will then just pass the DaemonSet you fetched